### PR TITLE
Add uncontrolled version of the component

### PR DESCRIPTION
**Pull request by zimme**
This will allow the input being uncontrolled and you can use `onSubmit` to just get the resulting value.

### CODE REVIEW

**Comment by vadimdemedes** [comment]
Sorry, I don't think we should support uncontrolled input, it's generally a bad practice and I would like to not encourage it :)

**Comment by zimme** [comment]
Sure I get the thinking behind that. However I do believe that's more the case when using redux/flux/etc architecture.

The "need" for allowing this to be an uncontrolled input came from the use-case where I'm not providing an initial value and I'm only interested in getting the value on submit and not as the user is typing. It's kinda annoying to have to manage my own state that won't even be used in my onSubmit handler as that provide the submitted value as an argument.

**Comment by zimme** [comment]
However, thinking about this a bit more... maybe the prop types check for value/onChange might need to be updated so that they are both required if any of them are provided. I can add this if this is to be merged.

**Comment by zimme** [comment]
or if we require `onChange` or `onSubmit` then this component will actually be controlled, just in 2 different ways.

**Comment by vadimdemedes** [comment]
Sorry for late reply, hammered with work lately.

Just got an idea, what if instead of modifying `<TextInput>`, we export another component called `<UncontrolledTextInput>`? It would simply wrap `<TextInput>` and handle the `value` and `onChange` props. It would keep `<TextInput>` controlled, keep the whole code clean and satisfy your use case at the same time :)

**Comment by zimme** [comment]
> Just got an idea, what if instead of modifying `<TextInput>`, we export another component called `<UncontrolledTextInput>`? It would simply wrap `<TextInput>` and handle the `value` and `onChange` props. It would keep `<TextInput>` controlled, keep the whole code clean and satisfy your use case at the same time :)

Sounds like a plan.

**Comment by zimme** [comment]
Done :tada: Created a new `UncontrolledTextInput` that I exported.

File: src/index.js [hunk 1]
```diff
@@ -169,3 +169,19 @@ export default class TextInputWithStdin extends PureComponent {
 		);
 	}
 }
+
+export class UncontrolledTextInput extends PureComponent {
```
Comment (vadimdemedes) [inline src/index.js]: You can simplify that with hooks :)

```jsx
export const UncontrolledTextInput = props => {
	const [value, setValue] = useState('');

	return (
		<TextInputWithStdin
			{...props}
			value={value}
			onChange={setValue)
		/>
	);
};
```

Could you also add `propTypes` to this new component?

File: readme.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ $ npm install ink-text-input
 import React from 'react';
 import {render, Box} from 'ink';
 import TextInput from 'ink-text-input';
+// import {UncontrolledTextInput as TextInput} from 'ink-text-input';
```
Comment (vadimdemedes) [inline readme.md]: Could you revert changes to the main example and a new "Uncontrolled Usage" section after "Props" section, which would document this component separately?

File: src/index.js (hunk 1 above)
Comment (zimme) [reply src/index.js]: I chose this approach as it aligns with the rest of the code and doesn't require react 16.8+

Comment (zimme) [reply src/index.js]: If I'm to add `propTypes` to this component, maybe we should add it to the original component too. What I mean is that `TextInputWithStdin` does not declare propTypes. So maybe we'll do another PR which adds propTypes to both `TextInputWithStdin` and `UncontrolledTextInput`?

Comment (vadimdemedes) [reply src/index.js]: Oh good idea, agree!

**Comment by vadimdemedes** [comment]
Nice work, @zimme, thank you! Changed readme a bit, but otherwise LGTM!

**Comment by vadimdemedes** [comment]
Released as `3.2.0`.

**Pull request merged**